tests/test_main.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2018 Shehriyar Qureshi <SShehriyar266@gmail.com>
+import os
+import sys
+
+# add renpy directory to path else main module isn't found
+current_directory = os.getcwd()
+sys.path.insert(0, current_directory)
+
+# thou shalt not move this artifact unless thou can without errors
+import main
+
+
+class TestRenpyPath:
+    def test_renpy_path_to_renpy_base(self):
+
+        # We're just doing what the function does and comparing the values.
+        # This test should fail if the function is changed and returns
+        # something else.
+
+        # we change sys.argv to mimimc as if renpy is being run
+        sys.argv.pop(1)
+        sys.argv[0] = 'renpy.py'
```

Comment:
Modifying sys.argv doesn't seem relevant here. If you modify sys.argv[0] to equal what you want, then you're not actually testing that main.path_to_renpy_base equals the result of sys.argv[0]. You're only testing that main.path_to_renpy_base equals what you think it should be.

Reply:
There are two things.

- We can leave `sys.argv` and check what the function does, which should be something similar i.e. it should give us a path to (folder where pytest executable is, in my case) but there is a problem with this approach:

    - There are functions that check for other folders inside of `renpy_base`. Creating tests for those functions will fail as we wouldn't be looking into the renpy directory.

The logical thing to do here in my opinion is to replicate normal renpy execution. That way other functions should work as intended and can be tested properly.

What I did here is not the best but it should do the trick for now. `sys.argv[0]` is replicating that `renpy.py` is being called from renpy directory.

There should be a better way to do this.

Reply:
The other functions aren't really **checking** if the folders were inside renpy directory. They are just appending the folders string to the path where script resides.

i.e. running renpy from renpy directory we get
`sys.argv = ['renpy.py']`
suppose path to `renpy.py` is `a/b/c/renpy.py`
the function returns path where script is placed, so it returns `a/b/c`

I updated the implementation to use whatever sys.argv we get (_which will be path to pytest_). This way:

`sys.argv[0] = 'x/y/z/pytest`
so the function should return `'x/y/z'`. We use `os.path.dirname` to get this path and use it
as our known output value.

This way we can just append the folders to `'x/y/z'.

i.e. `path_to_common` when supplied above path should return `'x/y/z/renpy/common'` and we can just append `/renpy/common` to above path and it will serve as our known output value. The test is going to fail if the function is changed in a way that it returns a different value.